chia/wallet/trading/offer.py

```diff
@@ -170,7 +170,7 @@ def conditions(self) -> Dict[Coin, List[Condition]]:
             for cs in self._bundle.coin_spends:
                 try:
                     cost, conds = cs.puzzle_reveal.run_with_cost(max_cost, cs.solution)
-                    max_cost -= cost
+                    max_cost = uint64(max_cost - cost)
```

Comment:
Similarly, what is the effect of raising here when less than 0 compared to existing behavior?